include/qore/intern/qore_thread_intern.h

```diff
@@ -363,6 +367,22 @@ DLLLOCAL QoreListNode* getCallStackList();
 #define popCall(x)
 #endif
 
+class ProgramCallContextHelper {
+public:
+    DLLLOCAL ProgramCallContextHelper(QoreProgram* new_pgm)
+        : pgm(new_pgm ? get_set_program_call_context(new_pgm) : reinterpret_cast<QoreProgram*>(-1)) {
```

Comment:
are u sure about reinterpret cast -1 into a pointer?

if I understand it correctly, you want to set new pgm and then return the old one, unless pgm is null, then you want to keep the original one. So cannot we simply do
```
pgm(new_pgm ? get_set_program_call_context(new_pgm) : get_program_call_context()) 
```
and then in destructor always set back the one that was there?

Reply:
as discussed yesterday